File: src/main/java/cz/vhromada/catalog/gui/picture/PictureDataPanel.java
package cz.vhromada.catalog.gui.picture;

import javax.swing.GroupLayout;
import javax.swing.ImageIcon;
import javax.swing.JLabel;

import cz.vhromada.catalog.entity.Picture;
import cz.vhromada.catalog.gui.common.AbstractDataPanel;

import org.springframework.util.Assert;

/**
 * A class represents panel with picture's data.
 *
 * @author Vladimir Hromada
 */
public class PictureDataPanel extends AbstractDataPanel<Picture> {

    /**
     * SerialVersionUID
     */
    private static final long serialVersionUID = 1L;

    /**
     * Label for picture
     */
    private final JLabel pictureData = new JLabel();

    /**
     * Creates a new instance of PictureDataPanel.
     *
     * @param picture picture
     * @throws IllegalArgumentException if picture is null
     */
    public PictureDataPanel(final Picture picture) {
        updateData(picture);

        pictureData.setFocusable(false);

        createLayout();
    }

    @Override
    protected void updateComponentData(final Picture data) {
        Assert.notNull(data, "picture");

        final byte[] picture = data.getContent();
        if (picture == null) {
            pictureData.setIcon(null);
        } else {
            pictureData.setIcon(new ImageIcon(picture));
        }
    }

    @Override
    protected String getCzWikiUrl() {
        throw new IllegalStateException("Getting URL to czech Wikipedia page is not allowed for pictures.");
    }

    @Override
    protected String getEnWikiUrl() {
        throw new IllegalStateException("Getting URL to english Wikipedia page is not allowed for pictures.");
    }

    @Override
    protected String getCsfdUrl() {
        throw new IllegalStateException("Getting URL to ČSFD page is not allowed for pictures.");
    }

    @Override
    protected int getImdbUrl() {
        throw new IllegalStateException("Getting URL to IMDB page is not allowed for pictures.");
    }

    @Override
    protected GroupLayout.Group getHorizontalLayoutWithComponents(final GroupLayout layout, final GroupLayout.Group group) {
        return group.addComponent(pictureData, HORIZONTAL_PICTURE_SIZE, HORIZONTAL_PICTURE_SIZE, HORIZONTAL_PICTURE_SIZE);
    }

    @Override
    protected GroupLayout.Group getVerticalLayoutWithComponents(final GroupLayout layout, final GroupLayout.Group group) {
        return group.addComponent(pictureData, VERTICAL_PICTURE_SIZE, VERTICAL_PICTURE_SIZE, VERTICAL_PICTURE_SIZE);
    }

}

File: src/main/kotlin/cz/vhromada/catalog/gui/picture/PictureDataPanel.kt
package cz.vhromada.catalog.gui.picture

import cz.vhromada.catalog.entity.Picture
import cz.vhromada.catalog.gui.common.AbstractDataPanel
import javax.swing.GroupLayout
import javax.swing.ImageIcon
import javax.swing.JLabel

/**
 * A class represents panel with picture's data.
 *
 * @author Vladimir Hromada
 */
class PictureDataPanel(picture: Picture) : AbstractDataPanel<Picture>() {

    /**
     * Label for picture
     */
    private val pictureData = JLabel()

    init {
        updateData(picture)

        pictureData.isFocusable = false

        createLayout()
    }

    override fun updateComponentData(data: Picture) {
        val picture = data.content
        if (picture == null) {
            pictureData.icon = null
        } else {
            pictureData.icon = ImageIcon(picture)
        }
    }

    override fun getCzWikiUrl(): String {
        error { "Getting URL to czech Wikipedia page is not allowed for pictures." }
    }

    override fun getEnWikiUrl(): String {
        error { "Getting URL to english Wikipedia page is not allowed for pictures." }
    }

    override fun getCsfdUrl(): String {
        error { "Getting URL to ČSFD page is not allowed for pictures." }
    }

    override fun getImdbUrl(): Int {
        error { "Getting URL to IMDB page is not allowed for pictures." }
    }

    override fun getHorizontalLayoutWithComponents(layout: GroupLayout, group: GroupLayout.Group): GroupLayout.Group {
        return group.addComponent(pictureData, HORIZONTAL_PICTURE_SIZE, HORIZONTAL_PICTURE_SIZE, HORIZONTAL_PICTURE_SIZE)
    }

    override fun getVerticalLayoutWithComponents(layout: GroupLayout, group: GroupLayout.Group): GroupLayout.Group {
        return group.addComponent(pictureData, VERTICAL_PICTURE_SIZE, VERTICAL_PICTURE_SIZE, VERTICAL_PICTURE_SIZE)
    }

}
